bootstrap.sh

```diff
@@ -1,117 +1,308 @@
+#set -e
+
+check_os
+install_packages_for_ansible_and_dependencies
+install_and_setup_homedir_files
+
+
+# Set secure permissions for the Vault file
+SECRET_FILE="${HOME}/ansible-easy-vpn/secret.yml"
+[[ -f "${SECRET_FILE}" ]] && {
+	echo
+	echo "WARNING:"
+	echo "${SECRET_FILE} already exists..."
+	echo
+}
+chmod 600 "${SECRET_FILE}"
+
+
+# Permissions are not critical with the CUSTOM_FILE
+# - secrets are not kept in this file
+CUSTOM_FILE="${HOME}/ansible-easy-vpn/custom.yml"
+[[ -f "${CUSTOM_FILE}" ]] && {
+	echo
+	echo "WARNING:"
+	echo "${CUSTOM_FILE} already exists..."
+	echo
+}
 
-# Check if we're running on an AWS EC2 instance
-set +e
-aws=$(curl -m 5 -s http://169.254.169.254/latest/meta-data/ami-id)
 
-if [[ "$aws" =~ ^ami.*$ ]]; then
-  aws=true
-else
-  aws=false
-fi
-set -e
-touch $HOME/ansible-easy-vpn/custom.yml
 
 
 
 clear
 echo "Welcome to ansible-easy-vpn!"
 echo
 echo "This script is interactive"
-echo "If you prefer to fill in the custom.yml file manually,"
+echo "If you prefer to fill in the ${CUSTOM_FILE} file manually,"
 echo "press [Ctrl+C] to quit this script"
 echo
 echo "Enter your desired UNIX username"
-read -p "Username: " username
-until [[ "$username" =~ ^[a-z0-9]*$ ]]; do
-  echo "Invalid username"
-  echo "Make sure the username only contains lowercase letters and numbers"
-  read -p "Username: " username
+username=
+until [[ ${username} =~ ^[a-z0-9]*$ && -n ${username} ]]; do
+	[[ -n ${username} ]] && echo "Invalid username"
+	echo "Make sure the username only contains lowercase letters and numbers"
+	read -r -p "Username: " username
 done
+# First write to CUSTOM_FILE, old content will be lost
+echo "username: \"${username}\"" |tee "${CUSTOM_FILE}"
 
-echo "username: \"${username}\"" >> $HOME/ansible-easy-vpn/custom.yml
 
 echo
 echo "Enter your user password"
 echo "This password will be used for Authelia login, administrative access and SSH login"
-read -s -p "Password: " user_password
-until [[ "${#user_password}" -lt 60 ]]; do
-  echo
-  echo "The password is too long"
-  echo "OpenSSH does not support passwords longer than 72 characters"
-  read -s -p "Password: " user_password
-done
-echo
-read -s -p "Repeat password: " user_password2
-echo
-until [[ "$user_password" == "$user_password2" ]]; do
-  echo
-  echo "The passwords don't match"
-  read -s -p "Password: " user_password
-  echo
-  read -s -p "Repeat password: " user_password2
+echo "Passwords longer trhan 72 bytes are not supported by OpenSSH private key format"
+echo " -- is this really true these days???"
+echo "Also, the password cannot be empty"
+while :
+do
+	user_password=
+	until [[ ${#user_password} -lt 73 && -n ${user_password} ]]; do
+		echo
+		[[ ${#user_password} -gt 72 ]] && echo "The password is too long"
+		read -s -r -p "Password: " user_password
+	done
+	echo
+	user_password2=
+	until [[ ${#user_password2} -lt 73 && -n ${user_password2} ]]; do
+		echo
+		[[ ${#user_password2} -gt 72 ]] && echo "The password is too long"
```

Comment:
As far as I understand, there's no need to check for password length for the repeated password. The user won't get to that point anyway if the first password is longer than 73 symbols, and if the repeated password is too long, it'll just throw the "Passwords don't match" error anyway